fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/FhirSchemaGenerator.java

```diff
@@ -704,6 +787,94 @@ protected void addCodeSystems(PhysicalDataModel model) {
+                .setIdentityColumn(   COMMON_TOKEN_VALUE_ID, Generated.ALWAYS)
+                .addIntColumn(               CODE_SYSTEM_ID,                          false)
+                .addVarcharColumn(              TOKEN_VALUE, MAX_TOKEN_VALUE_BYTES,   false)
+                .addUniqueIndex(IDX + tableName + "_TVCP", TOKEN_VALUE, CODE_SYSTEM_ID)
+                .addPrimaryKey(tableName + "_PK", COMMON_TOKEN_VALUE_ID)
+                .addForeignKeyConstraint(FK + tableName + "_CSID", schemaName, CODE_SYSTEMS, CODE_SYSTEM_ID)
+                .setTablespace(fhirTablespace)
+                .addPrivileges(resourceTablePrivileges)
+                .enableAccessControl(this.sessionVariable)
+                .build(pdm);
+
+        // TODO should not need to add as a table and an object. Get the table to add itself?
+        commonTokenValuesTable.addTag(SCHEMA_GROUP_TAG, FHIRDATA_GROUP);
+        pdm.addTable(commonTokenValuesTable);
+        pdm.addObject(commonTokenValuesTable);
+    }
+
+    /**
+     * Add the system-wide RESOURCE_TOKEN_REFS table which is used for
+     * _tag and _security search properties in R4 (new table
+     * for issue #1366 V0006 schema change). Replaces the
+     * previous TOKEN_VALUES table. All token values are now
+     * normalized in the COMMON_TOKEN_VALUES table
+     * @param pdm
+     * @return Table the table that was added to the PhysicalDataModel
+     */
+    public Table addResourceTokenRefs(PhysicalDataModel pdm) {
+
+        final String tableName = RESOURCE_TOKEN_REFS;
+
+        // logical_resources (0|1) ---- (*) resource_token_refs
+        Table tbl = Table.builder(schemaName, tableName)
+                .setVersion(FhirSchemaVersion.V0006.vid())
+                .setTenantColumnName(MT_ID)
+                .addIntColumn(       PARAMETER_NAME_ID,    false)
+                .addBigIntColumn(COMMON_TOKEN_VALUE_ID,     true) // support for null token value entries
+                .addBigIntColumn(  LOGICAL_RESOURCE_ID,    false)
+                .addIntColumn(          REF_VERSION_ID,     true) // for when the referenced value is a logical resource with a version
+                .addIndex(IDX + tableName + "_TVLR", COMMON_TOKEN_VALUE_ID, LOGICAL_RESOURCE_ID)
+                .addIndex(IDX + tableName + "_LRTV", LOGICAL_RESOURCE_ID, COMMON_TOKEN_VALUE_ID)
+                .addForeignKeyConstraint(FK + tableName + "_CTV", schemaName, COMMON_TOKEN_VALUES, COMMON_TOKEN_VALUE_ID)
+                .addForeignKeyConstraint(FK + tableName + "_LR", schemaName, LOGICAL_RESOURCES, LOGICAL_RESOURCE_ID)
+                .addForeignKeyConstraint(FK + tableName + "_PNID", schemaName, PARAMETER_NAMES, PARAMETER_NAME_ID)
+                .setTablespace(fhirTablespace)
+                .addPrivileges(resourceTablePrivileges)
+                .enableAccessControl(this.sessionVariable)
+                .build(pdm);
+
+        // TODO should not need to add as a table and an object. Get the table to add itself?
```

Comment:
the question made me smile... but makes sense

Reply:
Yeah, one of the things I'd like to clean up when we have some time